app/gui/src/controller/ide/desktop.rs

```diff
@@ -122,8 +136,9 @@ impl ManagingProjectAPI for Handle {
                 self.project_manager.create_project(&name, &version, &action).await?;
             let new_project_id = create_result.project_id;
             let project_mgr = self.project_manager.clone_ref();
-            let new_project = Project::new_opened(&self.logger, project_mgr, new_project_id);
-            self.current_project.set(new_project.await?);
+            let new_project =
+                Project::new_opened(&self.logger, project_mgr, new_project_id).await?;
```

Comment:
Are you able to refactor this line in such a way that it is not broken into 2 lines? You can create some additional variables if needed. Our code style tells about a strong preference towards single line expressions over multi-line ones.